src/Storages/Statistics/UniqStatistic.h

```diff
@@ -0,0 +1,61 @@
+#pragma once
+
+#include <Common/Arena.h>
+#include <Storages/Statistics/Statistics.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <AggregateFunctions/AggregateFunctionFactory.h>
+
+namespace DB
+{
+
+class UniqStatistic : public IStatistic
+{
+    std::unique_ptr<Arena> arena;
+    AggregateFunctionPtr uniq_collector;
+    AggregateDataPtr data;
+    UInt64 result;
```

Comment:
What does `result` represent? (please rename the variable or add a comment)
Is that the result of a cardinality estimation? If so, it seems risky and less beneficial to cache such results as estimation is supposed to be fast and can be done with many different values.